templates/web/pages/tag/tag.tt.html

```diff
@@ -48,6 +50,42 @@
             [% description %]
         </div>
     [% END %]
+    [% IF tagid.defined  %]
+        <!-- injecting facet-knowledge-panel -->
+        <div id="facet-knowledge-panel">
+            <h2 id="title"></h2>
+            <P id="facet-panels"></P>
+        </div>
+        <!-- Fetching facet knowledge panel -->
+        <script>
+            let facet_kp_url = "https://facets-kp.openfoodfacts.net/render-to-html?";
+            let params = "facet_tag=[% tagtype_name %]&value_tag=[% tagid %]&sec_facet_tag=[% tagtype2_name %]&sec_value_tag=[% tagid2 %]";
```

Comment:
tagtype_name is going to be something like "Categories" or "Catégories" in different language, you probably want "tagtype" instead, which be something like "categories" or "packaging_materials".

Same for tagid, canon_tagid is probably what you need.

Reply:
Yes , that is what I want but "tagtype" give categories" but I want "category" basically singular fom

Reply:
So for the tagtype, and for the taxonomies, we always use the plural in English as the id, I think it will be much easier if the facet knowledge panels keep that convention and use the same ids internally.

Reply:
for tagid2 what would canon_tagid look like I tried "canon_tagid2" seems its not working?

Reply:
Also, "canon_tagid2" seems not working

Reply:
I will make some changes in facet_kp that would solve this.